ui/src/chat/ChatUnreadAlerts.tsx

```diff
@@ -18,6 +23,16 @@ export default function ChatUnreadAlerts({
   whom,
 }: ChatUnreadAlertsProps) {
   const chatInfo = useChatInfo(whom);
+  const maybeWrit = useWrit(whom, chatInfo?.unread?.brief['read-id'] || '');
+  let replyToId: BigInteger = bigInt(0);
+  if (maybeWrit) {
+    const [_, writ] = maybeWrit;
+    const replyTo = writ.memo.replying || '';
+    replyToId =
+      replyTo !== ''
+        ? bigInt(replyTo.split('/')[1].replaceAll('.', ''))
+        : bigInt(0);
+  }
   const markRead = useCallback(() => {
     useChatState.getState().markRead(whom);
```

Comment:
I know this isn't what you were trying to solve but can you change this too. It will make the X work instantly. You'll probably have to import useChatStore

```suggestion
    useChatState.getState().markRead(whom);
    useChatStore.getState().read(whom);
```